app/components/order-inline-textarea-input.js

```diff
@@ -1,60 +1,75 @@
-import Ember from 'ember';
-import AutoResizableTextarea from './auto-resize-textarea';
-import AjaxPromise from 'stock/utils/ajax-promise';
+import Ember from "ember";
+import AutoResizableTextarea from "./auto-resize-textarea";
+import AjaxPromise from "stock/utils/ajax-promise";
 const { getOwner } = Ember;
 
 export default AutoResizableTextarea.extend({
-  previousValue: '',
+  previousValue: "",
   store: Ember.inject.service(),
   order: null,
-  orderCopy: Ember.computed.readOnly('order'),
+  orderCopy: Ember.computed.readOnly("order"),
 
   keyDown() {
     var value = this.element.value;
-    if( value.charCodeAt(value.length - 1) === 10 && event.which === 13) {
+    if (value.charCodeAt(value.length - 1) === 10 && event.which === 13) {
       return false;
     }
   },
 
   didInsertElement() {
-    Ember.$('.description-textarea').val(this.get('orderCopy.data.purposeDescription'));
+    Ember.$(".description-textarea").val(
+      this.get("orderCopy.data.purposeDescription")
+    );
   },
 
   focusOut() {
     var order = this.get("order");
-    var url = `/orders/${order.get('id')}`;
-    var key = this.get('name');
-    var value = this.attrs.value.value || '';
+    var url = `/orders/${order.get("id")}`;
+    var key = this.get("name");
+    var value = this.attrs.value.value || "";
     var orderParams = {};
-    orderParams[key] = this.get('value').trim() || '';
+    orderParams[key] = this.get("value").trim() || "";
     var element = this.element;
 
-    if (orderParams[key].toString() !== this.get('previousValue').toString().trim() && value !== ''){
-      var loadingView = getOwner(this).lookup('component:loading').append();
-      new AjaxPromise(url, "PUT", this.get('session.authToken'), { order: orderParams })
+    if (
+      orderParams[key].toString() !==
+        this.get("previousValue")
+          .toString()
+          .trim() &&
+      value !== ""
+    ) {
+      var loadingView = getOwner(this)
+        .lookup("component:loading")
+        .append();
+      new AjaxPromise(url, "PUT", this.get("session.authToken"), {
```

Comment:
I understand that it's hard to change, but that's a good example of a visual component with business logic. Which makes it very hard to re-use.